core/src/main/scala/com/rallyhealth/vapors/core/dsl/ZipExprSyntax.scala

```diff
@@ -0,0 +1,81 @@
+package com.rallyhealth.vapors.core.dsl
+
+import cats.{Align, Functor, FunctorFilter, Semigroupal}
+import com.rallyhealth.vapors.core.algebra.{CaptureP, Expr, ExprConverter, NonEmptyExprHList}
+import shapeless.ops.hlist.Tupler
+import shapeless.{::, Generic, HList, HNil}
+
+trait ZipExprSyntax {
+
+  def zip[F[_], V, M[_] : Align : Functor, E1, E2, P](
+    e1: Expr[F, V, M[E1], P],
+    e2: Expr[F, V, M[E2], P],
+  ): ZipExprHListWrapper[F, V, M, E1 :: E2 :: HNil, P] =
+    new ZipExprHListWrapper(e1 :: NonEmptyExprHList.tailK(e2))
+
+  def zip[F[_], V, M[_] : Align : Functor, E1, E2, E3, P](
+    e1: Expr[F, V, M[E1], P],
+    e2: Expr[F, V, M[E2], P],
+    e3: Expr[F, V, M[E3], P],
+  ): ZipExprHListWrapper[F, V, M, E1 :: E2 :: E3 :: HNil, P] =
+    new ZipExprHListWrapper(e1 :: e2 :: NonEmptyExprHList.tailK(e3))
+
+  def zip[F[_], V, M[_] : Align : Functor, E1, E2, E3, E4, P](
+    e1: Expr[F, V, M[E1], P],
+    e2: Expr[F, V, M[E2], P],
+    e3: Expr[F, V, M[E3], P],
+    e4: Expr[F, V, M[E4], P],
+  ): ZipExprHListWrapper[F, V, M, E1 :: E2 :: E3 :: E4 :: HNil, P] =
+    new ZipExprHListWrapper(e1 :: e2 :: e3 :: NonEmptyExprHList.tailK(e4))
+
+  def zip[F[_], V, M[_] : Functor : Semigroupal, E1, E2, E3, E4, E5, P](
+    e1: Expr[F, V, M[E1], P],
+    e2: Expr[F, V, M[E2], P],
+    e3: Expr[F, V, M[E3], P],
+    e4: Expr[F, V, M[E4], P],
+    e5: Expr[F, V, M[E5], P],
+  ): ZipExprHListWrapper[F, V, M, E1 :: E2 :: E3 :: E4 :: E5 :: HNil, P] =
+    new ZipExprHListWrapper(e1 :: e2 :: e3 :: e4 :: NonEmptyExprHList.tailK(e5))
+}
+
+final class ZipExprHListWrapper[F[_], V, M[_], L <: HList, P](private val exprHList: NonEmptyExprHList[F, V, M, L, P])
```

Comment:
I wonder if this should be more generically named than "zip". I could apply any higher-kinded operation to all the expressions.

Operations could be concatenation or permutations for example. In the case of concatenation, wouldn't the elements need to be of the same type? If so, this would not be a good interface for that outcome? In the case of permutations, this would be a good interface, but when would we ever need that operator?

If I try to come up with more generic names, I can think of `wrapEach` ("each" isn't really a common naming suffix for operations on collections), `wrapAll` (same comment about "all"... but maybe even worse), `wrapFoldable` (aligns with other spots like `.getFoldable`, except the types are not required to be `Foldable`), `group` (a new term, would not show up next to `wrap` in autocompletion, but short and fairly clear). I'm not happy with any of these more generic names. The would also require the method names in this class change to `.zip` for clarity.

Reply:
I suppose I could change `wrap` to `group` (so it would be `group(const(1), const("A")).asTuple` and call this method `groupEach`:

```
groupEach(const(List(1, 2, 3)), const("A", "B")).zippedToShortest.asTuples
// produces List((1, "A"), (2, "B"))
groupEach(const(List(1, 2, 3)), const("A", "B")).permutations.asTuples
// produces List((1, "A"), (1, "B"), (2, "A"), (2, "B"))
```